### DISCUSSION TITLE
LOW DETECTIONS but the SAME model with SAHI is better

### DISCUSSION THREAD

**Discussion by Misael777** [ultralytics/ultralytics#8089] [Q&A]
Hello I have a problem with the detection with yolo. My project works very well but when I tried to configure the conf threshold its begin to give me a low detections I tried to back my code how to was in the begining but it doesn´t works. I tried to make the inference with the SAHI library with exacly the same model that I use in my first program and it perform very well but with an elevate delay time processing the image so I can´t using for tracking objects in real time. I don´t know what is going on, I think that could be some of the configuration parameter of the predict function or in the moment to load the model. please help me.

This is my code with the SAHI Library
![WhatsApp Image 2024-02-07 at 4 29 30 PM](https://github.com/ultralytics/ultralytics/assets/63013799/79867838-8d13-4fd4-9336-79305a6feb47)

And this is my code with the Ultralytics library
![WhatsApp Image 2024-02-07 at 4 34 14 PM](https://github.com/ultralytics/ultralytics/assets/63013799/4e3db84e-b302-4c8f-8110-660fd3d455e6)
![WhatsApp Image 2024-02-07 at 4 38 03 PM](https://github.com/ultralytics/ultralytics/assets/63013799/61407ee5-9811-4c6b-b1dc-187187897b85)

**Comment by glenn-jocher** [comment]
@Misael777 it seems like you're experiencing a discrepancy in detection performance when using the same model with the SAHI library versus directly with the Ultralytics YOLOv8 library. This could be due to a number of factors, including differences in default confidence thresholds, non-maximum suppression (NMS) settings, or preprocessing steps.

To ensure consistent performance, please verify the following when using the Ultralytics YOLOv8 library:

1. **Confidence Threshold**: Ensure that the confidence threshold (`conf`) is set to the same value as used in SAHI. A lower threshold may result in more detections but also more false positives.

2. **NMS Threshold**: Check the IoU threshold (`iou`) used for NMS. This controls the overlap allowed between predicted bounding boxes.

3. **Preprocessing**: Confirm that the image preprocessing steps (resizing, normalization) are consistent between both methods.

4. **Model Loading**: Make sure you're loading the model correctly with the appropriate weights and configuration.

5. **Real-time Performance**: If you require real-time performance, consider optimizing your model with techniques like model pruning, quantization, or using a lighter model architecture.

Here's a basic example of how to set the confidence and IoU thresholds using the Ultralytics YOLOv8 library:

```python
from ultralytics import YOLO

# Load the model
model = YOLO('path/to/model.pt')

# Set the confidence and IoU thresholds
conf_threshold = 0.25  # Adjust to match SAHI's threshold
iou_threshold = 0.45   # Adjust to match SAHI's threshold

# Perform inference with custom thresholds
results = model.predict('path/to/image.jpg', conf=conf_threshold, iou=iou_threshold)
```

If you continue to experience issues, please provide more details about the specific configurations used in SAHI, and we can help you align the settings in the Ultralytics YOLOv8 library for consistent results. 🛠️
